pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -228,7 +228,12 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(ctx context.Context, cr
 			break
 		}
 
-		host, err := psmdb.MongoHost(ctx, r.client, cr, replset.Name, replset.Expose.Enabled, pod)
+		exposed := false
+		if cr.Spec.ClusterServiceDNSMode == api.DNSModeExternal {
+			exposed = replset.Expose.Enabled
+		}
+
+		host, err := psmdb.MongoHost(ctx, r.client, cr, replset.Name, exposed, pod)
 		if err != nil {
 			return api.AppStateError, fmt.Errorf("get host for pod %s: %v", pod.Name, err)
```

Comment:
If I understand correctly
1. We should update our CR.
2. We added one more unclear dependancy. 
Do we expose service or not now depends on `DNSModeExternal`. 
I don't like this since it makes our product even more obscure.